types/ContentType.php

```diff
@@ -220,6 +222,39 @@ public function duplicate(?array $loadData = null): CustomPage
             $newPage->updateAttributes(['url' => $newPage->url . '-' . $newPage->id]);
         }
 
+        $this->duplicateAttachments($newPage);
+
         return $newPage;
     }
+
+    /**
+     * Duplicate all attached files to new copied Custom Page
+     *
+     * @param CustomPage $newPage
+     * @return void
+     */
+    protected function duplicateAttachments(CustomPage $newPage): void
```

Comment:
@yurabakhtin That's quite a lot of new stuff in ContentType.
Wouldn't it make sense to create a 
`humhub\modules\custom_pages\service\DuplicateService`  that holds the logic and calls `beforeDuplicate` and `afterDuplicate` in `ContentType` if necessary?